internal/metamorph/processor.go

```diff
@@ -689,6 +727,9 @@ func (p *Processor) ProcessTransaction(ctx context.Context, req *ProcessorReques
 		return
 	}
 
+	// will be requesting transaction after ~2 seconds to get SEEN_ON_NETWORK status
```

Comment:
This operation may be blocking. Can we move it to the end of the processing? Does it make sense?

Reply:
What do you mean blocking? It's simple mutex lock nothing complicated here.

Reply:
Yes, but we lock the same collection in `StartCheckingTransactionsInNetwork`

Reply:
Sorry still confused. Yes we lock it in `StartCheckingTransactionsInNetwork` too but why is it a problem?